src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -2306,43 +2307,43 @@ void BytecodeInterpreter::run(interpreterState istate) {
         // QQQ Need to make this as inlined as possible. Probably need to split all the bytecode cases
         // out so c++ compiler has a chance for constant prop to fold everything possible away.
 
-        ConstantPoolCacheEntry* cache = cp->entry_at(index);
-        if (!cache->is_resolved((Bytecodes::Code)opcode)) {
+        ResolvedMethodEntry* entry = cp->resolved_method_entry_at(index);
+        if (!entry->is_resolved((Bytecodes::Code)opcode)) {
           CALL_VM(InterpreterRuntime::resolve_from_cache(THREAD, (Bytecodes::Code)opcode),
                   handle_exception);
-          cache = cp->entry_at(index);
+          entry = cp->resolved_method_entry_at(index);
```

Comment:
Do you actually need to lookup the entry again? I'm not really sure why the old code needed to do so.

Reply:
This is probably unnecessary. I will try removing it and see what the test results look like.